firefox-ios/Client/Frontend/Browser/BrowserViewController/State/BrowserViewControllerState.swift

```diff
@@ -201,6 +212,20 @@ struct BrowserViewControllerState: ScreenState, Equatable {
                 windowUUID: state.windowUUID,
                 browserViewType: state.browserViewType,
                 navigateToHome: showHomepage,
+                showQRcodeReader: state.showQRcodeReader,
+                microsurveyState: MicrosurveyPromptState.reducer(state.microsurveyState, action))
+        case GeneralBrowserActionType.showQRcodeReader:
+            let showQRcodeReader = action.showQRcodeReader ?? false
+            return BrowserViewControllerState(
+                searchScreenState: state.searchScreenState,
+                showDataClearanceFlow: state.showDataClearanceFlow,
+                toolbarState: state.toolbarState,
+                fakespotState: state.fakespotState,
+                toast: state.toast,
+                windowUUID: state.windowUUID,
+                browserViewType: state.browserViewType,
+                navigateToHome: state.navigateToHome,
+                showQRcodeReader: showQRcodeReader,
```

Comment:
From what I can see you should just be able to hardcode this to true and you don't need to include the showQRcodeReader field in the action.